beacon_node/network/src/router/processor.rs

```diff
@@ -916,62 +925,64 @@ impl<T: EthSpec> HandlerNetworkContext<T> {
         Self { network_send, log }
     }
 
-    pub fn disconnect(&mut self, peer_id: PeerId, reason: GoodbyeReason) {
-        warn!(
-            &self.log,
-            "Disconnecting peer (RPC)";
-            "reason" => format!("{:?}", reason),
-            "peer_id" => format!("{:?}", peer_id),
-        );
-        self.send_rpc_request(peer_id.clone(), RPCRequest::Goodbye(reason));
+    fn inform_network(&mut self, msg: NetworkMessage<T>) {
         self.network_send
-            .send(NetworkMessage::Disconnect { peer_id })
-            .unwrap_or_else(|_| {
-                warn!(
-                    self.log,
-                    "Could not send a Disconnect to the network service"
-                )
-            });
+            .send(msg)
+            .unwrap_or_else(|_| warn!(self.log, "Could not send message to the network service"))
     }
 
-    pub fn send_rpc_request(&mut self, peer_id: PeerId, rpc_request: RPCRequest<T>) {
-        // the message handler cannot send requests with ids. Id's are managed by the sync
-        // manager.
-        let request_id = 0;
-        self.send_rpc_event(peer_id, RPCEvent::Request(request_id, rpc_request));
+    // pub fn subscribe (
+    //     subscriptions: Vec<ValidatorSubscription>,
+    // ){}
+
+    // pub fn publish ( messages: Vec<PubsubMessage<T>> ){}
+
+    // pub fn propagate (
```

Comment:
I'm not sure the processor needs subscribe or publish, but potentially it could do propagate and shift it from the router to the processor